src/main.py

```diff
@@ -183,6 +184,66 @@ def to_volume(volume):
         volume = THIS_YEAR - 2011
     return int(volume)
 
+#
+# post processing
+#
+
+def do_section(element, fn):
+    "applies given function to each element in each section."
+    # sections may actually contain other sections. we treat the top-level 'body'
+    # attribute as just another section and recurse
+    if element['type'] == 'section': # recurse
+        element['content'] = [do_section(child, fn) for child in element['content']]
+    else:
+        element = fn(element)
+    return element
+
+def do_body(body_content, fn):
+    "applies given function to each non-section element in body content recursively"
+    return map(lambda element: do_section(element, fn), body_content)
+
+def do_body_for_type(body_content, type_list, fn):
+    def _fn(element):
+        return fn(element) if element['type'] in type_list else element
+    return do_body(body_content, _fn)
+
+def expand_any_videos(article):
+    "takes an existing video type struct as returned by elife-tools and fills it out with data from glencoe"
+    if 'body' not in article:
+        return article
+    msid = article['id']
+    body_content = article['body']
+    gc_data = glencoe.metadata(msid)
+    gc_id_str = ", ".join(gc_data.keys())
+
+    sources = {
+        'mp4': 'video/mp4; codecs="avc1.42E01E, mp4a.40.2"',
+        'webm': 'video/webm; codecs="vp8.0, vorbis"',
+        'ogv': 'video/ogg; codecs="theora, vorbis"',
+    }
+
+    def fn(video):
+        try:
+            vid = video['id']
+            ensure(vid in gc_data, "glencoe doesn't know %r, only %s" % (vid, gc_id_str))
```

Comment:
in the case of the testing environment, `msid` will be generated (something like `1234567890`) and `gc_data` should be `{}` as there are no video indexed with these generated keys on Glencoe. Will this part explode? Is there a way to make it skip not found videos (or should we try to load them with the original id?)